### ISSUE TITLE
Proposal: Turbo Stream apply to Template Tags

### ISSUE THREAD

**Issue by faqndo97** [hotwired/turbo#932]
## TL;DR

This proposal involves enhancing the functionality of Turbo streams to manipulate the content within a template tag. At present, Turbo streams are unable to access the content within a template tag due to the inability of `getElementById` and `querySelectorAll` to reach the `DocumentFragment`.

## Proposal

Consider the following scenario where we have a form that contains a template tag:

```erb
<%= form_tag url: some_path, model: ResourceA.new, data: { controller: "nested-form"} do |f| %>
  <%# TEMPLATES %>
    <template data-nested-form-target="nestedFields" data-fields-type="resourceB" >
      <%= f.fields_for :resource_b, ResourceB.new, child_index: 'TEMPLATE_RECORD' do |resource_b_form| %> 
        <%= resource_b_form.select :options, ... %>
      <% end %>
    </template>
  <%# END TEMPLATES %>

  <%= f.select :values, ... %>

  <%= f.fields_for :resource_b do |resource_b_form| %> 
    <%= resource_b_form.select :options, ... %>
  <% end %>

  <span data-action="click->nested-form#add_relationship" data-nested-form-field-type-param="resourceB">Add more resource B</span>
<% end %>
```

In this form, when a user clicks on the span, the `nested-form#add_relationship` is activated. This mechanism scans the template, takes the `innerHTML`, replaces `TEMPLATE_RECORD` with the current timestamp (`new Date().getTime()`) and appends it to the end of the form.

This functionality worked flawlessly until the requirement changed to "Update the options on `resource_b_form.select :options` when `f.select :values, changes`". Initially, I assumed this would be straightforward. By assigning a stimulus controller to `f.select :values`, I expected that any changes would prompt a server request using `request.js`. Subsequently, the Turbo stream would use a CSS selector to update all the options on the page. So far this continue working perfectly.

However, when a new `resourceB` was added, it did not contain the updated values from the Turbo stream. This issue arises because Turbo cannot reach the content inside a template tag, as it is a `DocumentFragment` which is not accessible by `getElementById` and `querySelectorAll`.

To address this, I propose that we should have the option to target these elements with Turbo streams. Template tags have numerous advantages: they are hidden by default, they don't initiate stimulus controllers, and they don't trigger anything else. The only disadvantage we face is the limited functionality of Turbo streams in this context.

As a temporary workaround, I added my own Turbo stream action and overwrote the `update_all` method. This method can receive an extra boolean parameter that determines whether to affect templates. If true, it will render both the Turbo stream update and my custom action to update templates.

If this feature is deemed beneficial, I am willing to assist in its development.

**Comment by dhh** [comment]
Feel free to explore a fix.
